lib/cli-engine.js

```diff
@@ -391,12 +392,14 @@ function getCacheFile(cacheFile, cwd) {
  */
 function CLIEngine(options) {
 
+    options = (options instanceof Object) ? assign(options) : {};
+    options.cwd = options.cwd || process.cwd();
+
     /**
      * Stored options for this instance
      * @type {Object}
      */
-    this.options = assign(Object.create(defaultOptions), options || {});
-
+    this.options = assign(defaultOptions, options);
```

Comment:
Fixing that resulted in [breaking another unit test](https://github.com/eslint/eslint/pull/3948#discussion_r48753826) that caused some confusion.

`Object.create(defaultOptions)` does not create an object with the same enumerable properties
```
> var a,b;
undefined
> a = { foo: 'bar' };
{ foo: 'bar' }
> b = Object.create(a);
{}
> b.foo
'bar'
> a.propertyIsEnumerable('foo')
true
> b.propertyIsEnumerable('foo')
false
```

The workaround is to to pass `Object.create(null)` as first argument to `object-assign`.